packages/client/components/MeetingSidebarTeamMemberStageItems.tsx

```diff
@@ -9,47 +9,66 @@ import MeetingSubnavItem from '../components/MeetingSubnavItem'
 import useGotoStageId from '../hooks/useGotoStageId'
 import MeetingSidebarPhaseItemChild from './MeetingSidebarPhaseItemChild'
+import {NewMeetingPhaseTypeEnum} from '../__generated__/DraggableReflectionCard_meeting.graphql'
+import {NavSidebar} from '../types/constEnums'
 
 const AvatarBlock = styled('div')({
   width: 32
 })
 
-const ScrollStageItems = styled('div')({
+const ScrollStageItems = styled('div')<{
+  showScrollbar: boolean
+}>(({showScrollbar}) => ({
   display: 'flex',
   flexDirection: 'column',
   height: '100%', // trickle down height for overflow
   // react-beautiful-dnd supports scrolling on 1 parent
   // this is where we need it, in order to scroll a long list
-  overflow: 'auto',
+  overflow: showScrollbar ? 'auto' : 'hidden',
   paddingRight: 8,
   width: '100%'
-})
+}))
 
 interface Props {
   gotoStageId: ReturnType<typeof useGotoStageId>
   handleMenuClick: () => void
   meeting: MeetingSidebarTeamMemberStageItems_meeting
+  phaseType: NewMeetingPhaseTypeEnum
+  maxSidebarChildrenHeight: number
 }
 
 const MeetingSidebarTeamMemberStageItems = (props: Props) => {
-  const {gotoStageId, handleMenuClick, meeting} = props
-  const {id: meetingId, facilitatorStageId, facilitatorUserId, localPhase, localStage} = meeting
-  const {phaseType} = localPhase
+  const {gotoStageId, handleMenuClick, meeting, phaseType, maxSidebarChildrenHeight} = props
+  const {
+    id: meetingId,
+    facilitatorStageId,
+    facilitatorUserId,
+    localPhase,
+    localStage,
+    phases
+  } = meeting
+  const {phaseType: localPhaseType} = localPhase
   const localStageId = (localStage && localStage.id) || ''
+  const atmosphere = useAtmosphere()
+  const {viewerId} = atmosphere
+  const isActivePhase = phaseType === localPhaseType
+  const isViewerFacilitator = viewerId === facilitatorUserId
+  const stages = phases.find((stage) => stage.phaseType === phaseType)?.stages
+  const stagesCount = stages?.length || 0
+  const maxHeight = NavSidebar.ITEM_HEIGHT * stagesCount
+  const childHeight = isActivePhase ? Math.min(maxSidebarChildrenHeight, maxHeight) : 0
+  const showScrollbar = maxHeight > maxSidebarChildrenHeight
   const gotoStage = (teamMemberId) => () => {
-    const teamMemberStage =
-      localPhase && localPhase.stages.find((stage) => stage.teamMemberId === teamMemberId)
+    const teamMemberStage = stages?.find((stage) => stage.teamMemberId === teamMemberId)
     const teamMemberStageId = (teamMemberStage && teamMemberStage.id) || ''
     gotoStageId(teamMemberStageId).catch()
     handleMenuClick()
   }
-  const atmosphere = useAtmosphere()
-  const {viewerId} = atmosphere
-  const isViewerFacilitator = viewerId === facilitatorUserId
+
   return (
-    <MeetingSidebarPhaseItemChild>
-      <ScrollStageItems>
-        {localPhase.stages.map((stage) => {
+    <MeetingSidebarPhaseItemChild height={childHeight}>
+      <ScrollStageItems showScrollbar={showScrollbar}>
+        {stages!.map((stage) => {
```

Comment:
When transitioning from checkin or updates phase to agendaitems phase, the localPhase will be agendaitems during the 300ms transition when the phase is closing. As we still want to show the checkin or updates children during those 300ms, I'm using the stages instead of the localPhase